### ISSUE TITLE
Support to writing back the yaml to files

### ISSUE THREAD

**Issue by Raffo** [kubernetes-sigs/kustomize#55]
Hi, thanks for this great project, I was really looking for something like that and I am excited for the future developments! 

I was thinking if it would make sense from your point of view to have the option to write the output yamls to files following the original names instead of STDOUT. 
My use cause would be to use them in a deploy script that knowns about some conventions, i.e. I don't want to apply a deployment and thus replacing 100% of my instances, but have a blue/green approach. Such thing should be not part of `kustomize` and having the file written back to disk would be enough for me to build a logic around the generated file. 

Do you mind sharing your opinion on that proposal?

**Comment by monopole** [comment]
kustomize in general wants to support a fork/rebase workflow in which one has a fork of someone else's repo, and occasionally does `git rebase` to capture changes which then become part of your workflow. So we don't want to overwrite the base (would could be just one big YAML file with many resources).

That's why there's currently no attempt to retain any kind of mapping between objects and the file they came from (or the order in which the objects appeared in those files).

> i.e. I don't want to apply a deployment and thus replacing 100% of my instances,
> but have a blue/green approach. 

Could you take a look at [hello world example](https://github.com/kubernetes-sigs/kustomize/tree/master/examples/helloWorld#compare-overlays) but replace (_staging_, _production_) with (_blue_, _green_), and describe why that pattern does or does not work
for your case?

**Comment by sethpollack** [comment]
@monopole what if we write to a release dir instead of overwriting the base? You can add a release dir in the kustomization file where all the modified resources would be written to.

**Comment by monopole** [comment]
Yep, apply label filtering works, but it's an imperative approach in that the label filter isn't necessary captured in version control (unless you store a script obviously).

I'd like a clearer explanation of what problem @Raffo wants to address with blue/green deployment.  One can use different kustomization files to involve different sets of resources (that what overlays are).

**Comment by geekflyer** [comment]
Hi @monopole and co., I also would like to request support to writing the kustomize output to files instead of only stdout. The reason I'd like this is for debugging the output to verify if a particular resource was correctly "kustomized". Imagine I have 10 resources (in my current use case it's already 9) managed by the same kustomization. In this case I'd like to write all the kustomized resources to a separate directory and then inspect them in one-by-one in my editor. I know I could write all of them to a single file, but when having 10 or more resources this single file becomes already gigantic and it's hard to navigate around.

So in summary I think it would be really helpful to have the ability to write the kustomized resources as seperate files to a directory.

**Comment by Liujingfang1** [comment]
We will add `-o` to write the output to a file. #245

**Comment by kunickiaj** [comment]
Doesn't seem like #245 addresses this issue, which is similar to #227 but for separating each resource.

for example:

```
kustomize build production | csplit - --prefix='production.' --suffix-format='%03d.yaml' --elide-empty-files --keep-files /---/ '{*}'
```
Is one workaround to splitting the resulting output into separate files, but all context is lost. Would be nice to have something similar to `helm template` where a comment is added to each resource indicating where it originated from.

**Comment by pdefreitas** [comment]
To do a blue/green deployments with kustomize we can seperate bases (between deployments and other kubernetes objects) and adding 2 overlays: one for the blue step, another for the green step.
This seems to satisfy the current use case I am experiencing, but it would be nice it we could control the outputted content/files a little better.
